riptide-core/src/main/java/org/zalando/riptide/UrlResolution.java

```diff
@@ -0,0 +1,58 @@
+package org.zalando.riptide;
+
+import com.google.common.base.Strings;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+
+import static com.google.common.base.Strings.nullToEmpty;
+
+// TODO introduce UrlResolutionStrategy interface in the future, if needed
+public enum UrlResolution {
+
+    /**
+     * Resolves a given relativce URI by following the rules defined in RFC 3986.
+     *
+     * @see <a href="https://tools.ietf.org/html/rfc3986">RFC 3986</a>
+     * @see URI#resolve(URI)
+     */
+    RFC {
+        @Override
+        URI resolve(final URI baseUrl, final URI uri) {
+            return baseUrl.resolve(uri);
+        }
+    },
+
+    /**
+     * Resolves a given relative URI by appending its path to and merging its query parameters with the given base URL.
+     *
+     * @see UriComponentsBuilder#pathSegment(String...)
```

Comment:
This one is not used anymore, isn't it?
Also I am still thinking of cornercases like `http://example.com` + `/` and `http://example.com/` + `/` :)